src/main/java/com/minecolonies/core/entity/pathfinding/navigation/IMinecoloniesPather.java

```diff
@@ -0,0 +1,13 @@
+package com.minecolonies.core.entity.pathfinding.navigation;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.entity.ai.navigation.PathNavigation;
+
+/**
+ * This interface should be applied only to classes whose navigation can be safely cast to MinecoloniesAdvancedPathNavigate
+ */
+public interface IMinecoloniesPather
+{
+    public PathNavigation getNavigation();
```

Comment:
since vanilla already declares this method it'll have problems with the unmapped method name in 1.20.1 forge, 
since the intention here is to get the minecolonies navigator why not change this to a custom
getMcolNavigator()
function? which the entities then implement.
Same issues with the blockpos, though you can just remove that and instead use a combined type for the Utilities param, e.g. T <T extends Mob & IThreatTableEntity>
so you automatically get the Interface & normal entity methods